operator/src/test/java/org/keycloak/operator/KeycloakDeploymentE2EIT.java

```diff
@@ -0,0 +1,138 @@
+package org.keycloak.operator;
+
+import io.fabric8.kubernetes.api.model.EnvVarBuilder;
+import io.quarkus.logging.Log;
+import io.quarkus.test.junit.QuarkusTest;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.keycloak.operator.utils.CRAssert;
+import org.keycloak.operator.v2alpha1.crds.Keycloak;
+import org.keycloak.operator.v2alpha1.crds.KeycloakStatusCondition;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.keycloak.operator.utils.K8sUtils.deployKeycloak;
+import static org.keycloak.operator.utils.K8sUtils.getDefaultKeycloakDeployment;
+import static org.keycloak.operator.utils.K8sUtils.waitForKeycloakToBeReady;
+
+@QuarkusTest
+public class KeycloakDeploymentE2EIT extends ClusterOperatorTest {
+    @Test
+    public void given_ClusterAndOperatorRunning_when_KeycloakCRCreated_Then_KeycloakStructureIsDeployedAndStatusIsOK() {
+        try {
+            // CR
+            Log.info("Creating new Keycloak CR example");
+            var kc = getDefaultKeycloakDeployment(k8sclient);
+            var deploymentName = kc.getMetadata().getName();
+            deployKeycloak(k8sclient, kc, true);
+
+            // Check Operator has deployed Keycloak
+            Log.info("Checking Operator has deployed Keycloak deployment");
+            assertThat(k8sclient.apps().deployments().inNamespace(namespace).withName(deploymentName).get()).isNotNull();
+
+            // Check Keycloak has correct replicas
+            Log.info("Checking Keycloak pod has ready replicas == 1");
+            assertThat(k8sclient.apps().deployments().inNamespace(namespace).withName(deploymentName).get().getStatus().getReadyReplicas()).isEqualTo(1);
+
+            // Delete CR
+            Log.info("Deleting Keycloak CR and watching cleanup");
+            k8sclient.resources(Keycloak.class).delete(kc);
+            Awaitility.await()
+                    .untilAsserted(() -> assertThat(k8sclient.apps().deployments().inNamespace(namespace).withName(deploymentName).get()).isNull());
+        } catch (Exception e) {
+            savePodLogs();
+            throw e;
+        }
+    }
+
+    @Test
+    public void testImageChange() {
+        try {
+            var kc = getDefaultKeycloakDeployment(k8sclient);
+            kc.getSpec().setImage("quay.io/keycloak/non-existing-keycloak");
+
+            deployKeycloak(k8sclient, kc, false);
+            waitForKcToFail(kc);
```

Comment:
This check is not quite enough to tell that it's failing for `ImagePullBackoff`, either:
 - remove the test at all and tackle it in https://github.com/keycloak/keycloak/issues/10099
 - check for the specific condition

Reply:
👍